Gruntfile.js

```diff
@@ -169,17 +169,37 @@ module.exports = function(grunt) {
         }
       }
     },
-      uglify: { 
-          options: { 
-              compress: true, 
-              mangle: true, 
-              sourceMap: false
-          }, 
-          target: { 
-              src: './dist/browser/js/ion-bundle.js',
-              dest: './dist/browser/js/ion-bundle.min.js',
-          }
+    uglify: {
+        options: {
+            compress: true,
+            mangle: true,
+            sourceMap: false
+        },
+        target: {
+            src: './dist/browser/js/ion-bundle.js',
+            dest: './dist/browser/js/ion-bundle.min.js',
+        }
+    },
+    mochaTest: {
+      test: {
+        options: {
+          ui: 'mocha-typescript',
+          require: [
+              'ts-node/register',
+              'source-map-support/register',
+              'nyc'
+          ],
+          noFail: false // Run all tests even if an early test fails
+        },
+        src: ['test/**/*.ts']
       }
+    },
+    shell: {
```

Comment:
The `nyc` command is the CLI interface to `Instanbul`, our code coverage tool. There's not an easy way to mix Grunt + Typescript + Mocha + nyc, so I've opted to simply invoke it as a child process.

Reply:
is this the right time to upgrade our grunt dependency as well?

Reply:
Should we do the same approach of running the unit tests by just invoking it rather than using the plugin?  In other words, if this is our test target from npm, does the grunt integration matter?

Reply:
@wesboyt Our package.json is pointed at `grunt` v1.0.4, which looks to be the latest release. Is that what you were referring to? I'm going to leave any package upgrades for another PR, but let me know what you were thinking.

@almann It's probably not in keeping with the spirit of `grunt`, but it sure seems easier to me. It would cut our dependency list down substantially and eliminate some config duplication. One downside is that it can't be done "natively"--you have to add a grunt package to run child processes without reinventing the wheel, and there are quite a few to choose from. (`grunt-exec`, `grunt-shell`, `grunt-run`, etc.) These have different properties for configuration, color pass-through, and presumably overhead.